src/code.cpp

```diff
@@ -0,0 +1,14 @@
+#include <Rcpp.h>
+#include "hiperglm_types.h"
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+
+VectorXd QR_solver_eigen(MatrixXd A, VectorXd b){
+  if (A.rows() != b.size()) {
+    Rcpp::stop("Incompatible matrix and vector dimension.");
+  }
+  HouseholderQR<MatrixXd> qr(A);
+  qr = qr.compute(A);
```

Comment:
```suggestion
```
I don't think you need this line, though maybe with your particular version it was needed. .solve() computes the qr itself.